bluefog/torch/optimizers.py

```diff
@@ -335,42 +352,42 @@ def __init__(self, params, model, communication_type, num_steps_per_communicatio
 
     def _register_hooks(self):
         for model in self._models:
-            for parent_name, layer in _named_leaf_module(model):
-                layer.register_forward_hook(self._make_hook(parent_name))
+            model.register_forward_hook(self._make_hook())
+            for _, layer in _named_leaf_module(model):
+                #layer.register_forward_hook(self._make_hook(parent_name))
                 for _, p in layer.named_parameters():
                     self._requires_update.add(p)
```

Comment:
why not use model.named_parameters() to set the self. _requires_update

Reply:
Done.